pendulum_msgs/msg/JointState.msg

```diff
@@ -0,0 +1,3 @@
+float64 position
```

Comment:
> I don't think we can make the complete pipeline preallocated here

I don't think it's necessary to preallocate the entire pipeline.

> I think Adolfo is suggesting a way to lock dynamically sized messages during the startup stage of the node lifecycle, so that you can use a preallocated, statically sized buffer for the message during execution. Is that accurate? It seems like this is useful when the size of the messages becomes known when the program starts, but does not change during real-time execution.

Yes, that's what I meant. And that's one of the reasons why node lifecycle is important. I think it's reasonable to assume that message and other data structure sizes can be determined (and pre-allocated) at initialization/configure time, which is non real-time. The active/running state would incur no further allocations. I find it also reasonable to expect a node reconfiguration (something like stop→cleanup→configure→start) to account for a problem size change.

I don't doubt there are very legitimate usecases out there for hard-coded message upper bounds, but I've so far not had the need for them. On the other hand, I can't imagine generic robot control software without (pre-allocated) dynamically sized messages.